<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>ShaderWidget</class>
 <widget class="QWidget" name="ShaderWidget">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>360</width>
    <height>365</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="windowTitle">
   <string>Form</string>
  </property>
  <property name="layoutDirection">
   <enum>Qt::LeftToRight</enum>
  </property>
  <layout class="QHBoxLayout" name="horizontalLayout">
   <property name="spacing">
    <number>2</number>
   </property>
   <property name="leftMargin">
    <number>2</number>
   </property>
   <property name="topMargin">
    <number>2</number>
   </property>
   <property name="rightMargin">
    <number>2</number>
   </property>
   <property name="bottomMargin">
    <number>2</number>
   </property>
   <item>
    <widget class="QTabWidget" name="tabWidget">
     <property name="tabPosition">
      <enum>QTabWidget::South</enum>
     </property>
     <property name="currentIndex">
      <number>1</number>
     </property>
     <widget class="QWidget" name="tabShaders">
      <attribute name="title">
       <string>Shaders</string>
      </attribute>
      <layout class="QHBoxLayout" name="horizontalLayout_2">
       <property name="spacing">
        <number>2</number>
       </property>
       <property name="leftMargin">
        <number>0</number>
       </property>
       <property name="topMargin">
        <number>0</number>
       </property>
       <property name="rightMargin">
        <number>0</number>
       </property>
       <property name="bottomMargin">
        <number>0</number>
       </property>
       <item>
        <layout class="QVBoxLayout" name="verticalLayout_2">
         <property name="spacing">
          <number>0</number>
         </property>
         <property name="leftMargin">
          <number>0</number>
         </property>
         <property name="topMargin">
          <number>0</number>
         </property>
         <item>
          <widget class="QToolBar" name="toolBarShaders">
           <property name="floatable">
            <bool>false</bool>
           </property>
           <addaction name="actionAddShader"/>
           <addaction name="actionDeleteShader"/>
           <addaction name="actionPreview"/>
           <addaction name="actionCompileOutput"/>
           <addaction name="actionUniforms"/>
          </widget>
         </item>
         <item>
          <widget class="QListWidget" name="listWidget">
           <property name="maximumSize">
            <size>
             <width>200</width>
             <height>16777215</height>
            </size>
           </property>
          </widget>
         </item>
         <item>
          <widget class="NovelTeaWidget" name="preview" native="true">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="minimumSize">
            <size>
             <width>0</width>
             <height>100</height>
            </size>
           </property>
           <property name="maximumSize">
            <size>
             <width>200</width>
             <height>16777215</height>
            </size>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QtTreePropertyBrowser" name="propertyBrowser" native="true">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="minimumSize">
            <size>
             <width>0</width>
             <height>100</height>
            </size>
           </property>
           <property name="maximumSize">
            <size>
             <width>200</width>
             <height>16777215</height>
            </size>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QVBoxLayout" name="verticalLayout_3">
         <property name="spacing">
          <number>2</number>
         </property>
         <property name="rightMargin">
          <number>0</number>
         </property>
         <property name="bottomMargin">
          <number>0</number>
         </property>
         <item>
          <widget class="ScriptEdit" name="scriptEdit">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
             <horstretch>0</horstretch>
             <verstretch>3</verstretch>
            </sizepolicy>
           </property>
           <property name="lineWrapMode">
            <enum>QPlainTextEdit::NoWrap</enum>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPlainTextEdit" name="textErrorLog">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
             <horstretch>0</horstretch>
             <verstretch>1</verstretch>
            </sizepolicy>
           </property>
           <property name="font">
            <font>
             <family>DejaVu Sans Mono</family>
            </font>
           </property>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="tabTextures">
      <attribute name="title">
       <string>Textures</string>
      </attribute>
      <layout class="QHBoxLayout" name="horizontalLayout_3">
       <property name="spacing">
        <number>2</number>
       </property>
       <property name="leftMargin">
        <number>0</number>
       </property>
       <property name="topMargin">
        <number>0</number>
       </property>
       <property name="rightMargin">
        <number>0</number>
       </property>
       <property name="bottomMargin">
        <number>0</number>
       </property>
       <item>
        <layout class="QVBoxLayout" name="verticalLayout">
         <property name="spacing">
          <number>2</number>
         </property>
         <property name="bottomMargin">
          <number>0</number>
         </property>
         <item>
          <widget class="QToolBar" name="toolBarTextures">
           <property name="floatable">
            <bool>false</bool>
           </property>
           <addaction name="actionAddTexture"/>
           <addaction name="actionRemoveTexture"/>
          </widget>
         </item>
         <item>
          <widget class="QListWidget" name="listTextures">
           <property name="iconSize">
            <size>
             <width>100</width>
             <height>100</height>
            </size>
           </property>
           <property name="resizeMode">
            <enum>QListView::Adjust</enum>
           </property>
           <property name="spacing">
            <number>10</number>
           </property>
           <property name="viewMode">
            <enum>QListView::IconMode</enum>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QFormLayout" name="formLayout_2">
         <property name="rightMargin">
          <number>0</number>
         </property>
         <item row="0" column="0">
          <widget class="QLabel" name="label_3">
           <property name="text">
            <string>TextLabel</string>
           </property>
          </widget>
         </item>
         <item row="0" column="1">
          <widget class="QComboBox" name="comboBox"/>
         </item>
        </layout>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="tabDefaults">
      <attribute name="title">
       <string>Defaults</string>
      </attribute>
      <layout class="QFormLayout" name="formLayout">
       <property name="fieldGrowthPolicy">
        <enum>QFormLayout::AllNonFixedFieldsGrow</enum>
       </property>
       <item row="0" column="0">
        <widget class="QLabel" name="label">
         <property name="text">
          <string>Background</string>
         </property>
        </widget>
       </item>
       <item row="0" column="1">
        <widget class="QComboBox" name="comboShaderBackground"/>
       </item>
       <item row="1" column="0">
        <widget class="QLabel" name="label_2">
         <property name="text">
          <string>Post Process</string>
         </property>
        </widget>
       </item>
       <item row="1" column="1">
        <widget class="QComboBox" name="comboShaderPostProcess"/>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
  </layout>
  <action name="actionAddShader">
   <property name="icon">
    <iconset theme="add">
     <normaloff/>
    </iconset>
   </property>
   <property name="text">
    <string>Add Shader</string>
   </property>
  </action>
  <action name="actionDeleteShader">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="icon">
    <iconset theme="remove">
     <normaloff/>
    </iconset>
   </property>
   <property name="text">
    <string>Delete Shader</string>
   </property>
  </action>
  <action name="actionPreview">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset theme="desktop">
     <normaloff/>
    </iconset>
   </property>
   <property name="text">
    <string>Preview</string>
   </property>
  </action>
  <action name="actionUniforms">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset theme="document-properties">
     <normaloff/>
    </iconset>
   </property>
   <property name="text">
    <string>Uniforms</string>
   </property>
  </action>
  <action name="actionCompileOutput">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset theme="help">
     <normaloff/>
    </iconset>
   </property>
   <property name="text">
    <string>Compile Output</string>
   </property>
  </action>
  <action name="actionAddTexture">
   <property name="icon">
    <iconset theme="add">
     <normaloff/>
    </iconset>
   </property>
   <property name="text">
    <string>Add Texture</string>
   </property>
  </action>
  <action name="actionRemoveTexture">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="icon">
    <iconset theme="remove">
     <normaloff/>
    </iconset>
   </property>
   <property name="text">
    <string>Remove Texture</string>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>ScriptEdit</class>
   <extends>QPlainTextEdit</extends>
   <header>Widgets/ScriptEdit.hpp</header>
  </customwidget>
  <customwidget>
   <class>NovelTeaWidget</class>
   <extends>QWidget</extends>
   <header>Widgets/NovelTeaWidget.hpp</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>QtTreePropertyBrowser</class>
   <extends>QWidget</extends>
   <header>Widgets/QtPropertyBrowser/qttreepropertybrowser.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
